src/launch/launcher.rs

```diff
@@ -0,0 +1,37 @@
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::launch::linux::ioctl::*;
+use crate::launch::types::*;
+
+use std::io::Result;
+use std::os::unix::io::{AsRawFd, RawFd};
+
+pub struct New;
+
+struct Fd<'a>(&'a dyn AsRawFd);
+
+impl AsRawFd for Fd<'_> {
+    fn as_raw_fd(&self) -> RawFd {
+        self.0.as_raw_fd()
+    }
+}
+
+pub struct Launcher<'a, T> {
+    state: T,
+    kvm: Fd<'a>,
+    sev: Fd<'a>,
+}
```

Comment:
```suggestion
pub struct Launcher<'a, T> {
    state: T,
    kvm: &'a dyn AsRawFd,
    sev: &'a dyn AsRawFd,
}
```

Alternatively:

```suggestion
pub struct Launcher<'a, T, U: AsRawFd, V: AsRawFd> {
    state: T,
    kvm: &'a U,
    sev: &'a V,
}
```

You might also want to hold a mutable reference just to enforce that nobody else is using the file during attestation.